numpy/random/mtrand/generate_randint_helpers.py

```diff
@@ -0,0 +1,152 @@
+"""
+Helper file for generating the necessary helper functions
+for each `dtype` in `np.random.randint`. The function
+`gen_randint_helpers` is called right before Cythonization
+in NumPy setup.
+
+"""
+
+# npy_dt : Desired C API data type
+# npy_udt : Unsigned version of desired C
+#           API data type (could be identical)
+# npy_dt : Desired Python API data type
+
+template = """
+def _rand_{npy_dt}(low, high, size, rngstate):
+    \"\"\"
+    _rand_{npy_dt}(low, high, size, rngstate)
+
+    Return an array of random np.{np_dt} integers between ``lo`` and ``hi``
+    inclusive for each ``lo``, ``hi`` pair in ``low`` and ``high``, where
```

Comment:
you might be able to do this with fused types http://docs.cython.org/src/userguide/fusedtypes.html

only trick is you need to pass a c pointer to function based on the fused type, but you could do that with a lookup table or if-else (or might be a better way)